File: problems/2576_find_the_maximum_number_of_marked_indices/solution.py
from typing import List


class Solution:
    def maxNumOfMarkedIndices(self, nums: List[int]) -> int:
        nums.sort()
        n = len(nums)
        ans = 0
        j = n // 2
        i = 0
        while i < n // 2 and j < n:
            if 2 * nums[i] <= nums[j]:
                ans += 2
                i += 1
            j += 1
        return ans

File: problems/2576_find_the_maximum_number_of_marked_indices/solution.cpp
#include <vector>
#include <string>
#include <unordered_map>
#include <unordered_set>
#include <algorithm>
#include <climits>
#include <queue>
#include <stack>
#include <cmath>
#include <numeric>
#include <set>
#include <map>
using namespace std;

class Solution {
public:
    int maxNumOfMarkedIndices(vector<int>& nums) {
        sort(nums.begin(), nums.end());
        int n = nums.size(), ans = 0, i = 0;
        for (int j = n/2; j < n && i < n/2; j++)
            if (2 * nums[i] <= nums[j]) { ans += 2; i++; }
        return ans;
    }
};
